import-service/handlers/importFileParser.js

```diff
@@ -0,0 +1,36 @@
+const { S3 } = require('aws-sdk');
+const csv = require('csv-parser');
+const { BUCKET_NAME, S3_OPTIONS } = require('../utils/constants');
+const response = require('../utils/response');
+
+module.exports = async function importFileParser(event) {
+  try {
+    const s3 = new S3(S3_OPTIONS);
+    const objects = await Promise.all(
+      event.Records.map(async (record) => {
+        const Bucket = BUCKET_NAME;
+        const { key: Key } = record.s3.object;
+        // create stream
+        const stream = await s3.getObject({ Bucket, Key }).createReadStream().pipe(csv({
+          headers: ['title', 'description', 'price'],
+        }));
+        // retrieve data
+        const list = [];
+        for await (const data of stream) { // eslint-disable-line no-restricted-syntax
+          console.log('-->', Key, data);
+          list.push(data);
+        }
+        console.log('-- LIST', Key, list);
```

Comment:
I think storing chunks of streamed data in the array is not a good idea. Using streams helps us to process data bits by bits, so the memory footprint of the app would be tiny. But your implementation defeats its purpose by storing the whole file in memory. Imagine, for example, processing a CSV file over 3 gigabytes, this lambda function would be terminated due to memory overflow.

https://aws.amazon.com/about-aws/whats-new/2017/11/aws-lambda-doubles-maximum-memory-capacity-for-lambda-functions/#:~:text=AWS%20Lambda%20Doubles%20Maximum%20Memory%20Capacity%20for%20Lambda%20Functions,-Posted%20On%3A%20Nov&text=You%20can%20now%20allocate%203008MB,to%20your%20functions%20was%201536MB.

Reply:
Thank you very much)
You are absolutely correct about the streams and storing received data in the array. I definitely will fix it with the next update to the service